sdk/program/PROJECT.md

```diff
@@ -0,0 +1,138 @@
+## Future of the solana-program
+
+The Solana Program SDK exists in the Agave validator monorepo. This has been
+convenient for validator developers, but its presence in the monorepo has
+sometimes hindered on-chain program developers. Since its release cycle is tied
+to the Agave validator, it is not possible to easily release breaking changes.
+
+The issue for on-chain developers has been brought up or debated in issues like:
+
+* https://github.com/solana-labs/solana/issues/32503
+* https://github.com/solana-labs/solana/issues/32625
+* https://github.com/anza-xyz/agave/issues/20
+
+To give on-chain program developers a better experience, it's time to define
+some priorities and a plan to make things better.
+
+## Concepts for the Program SDK
+
+Here are the priorities for the future of the SDK, in order of importance.
+
+1. Secure
+2. Respect semantic versioning
+3. On-chain program developers come first
+4. Possible to build a new program quickly and easily
+5. Sensible defaults, but allow customization
+6. Minimal dependencies
+7. Kitchen sinks only re-export functionality
+8. Default to new crates or files to bundle functionality
+9. If new functionality can't be done in a new crate or file, use features
```

Comment:
imo we need to be _very_ careful with what is featurized to avoid testing complexity